surveymonkey/api_surveymonkey.py

```diff
@@ -86,13 +98,23 @@ def get_surveys(self, **kwargs):
         """
         Returns a list of surveys owned or shared with the authenticated user.
         """
+        cache_key = "{}-{}-{}".format(SURVEY_MONKEY_API_TAG, 'all_surveys', self.client_id)
+        all_surveys_data = cache.get(cache_key)
+
+        if self.surveymonkey_api_cache_duration == 0:
```

Comment:
https://docs.djangoproject.com/en/1.11/topics/cache/#basic-usage I don't understand the question. @andrey-canon

Reply:
is the line porpoise to delete the cache is the `surveymonkey_api_cache_duration` is updated ?

Reply:
This is the "kill switch" to delete the cache and get the responses again. @andrey-canon

Reply:
if you  do that you are deleting the cache for all the xblocks, for example if you have an xblock `a` with `surveymonkey_api_cache_duration = 154112454` and an xblock `b` with  `surveymonkey_api_cache_duration = 0` the xblock b is deleting the data  for  the xblock a  
 when the user specific a time of   154112454, that is not a good design to  affect another xblock with the  other xblock settings.

Reply:
Yes, but this is the method to get the all surveys, so, the only thing that will do is get the response again for all the surveys. https://developer.surveymonkey.com/api/v3/#surveys

Reply:
in this case it doesn't care, but how ugly!

Reply:
Have you a better idea to say ugly? @andrey-canon

Reply:
Only two
1. dont create a setting that will be not respected.
2. dont add the option to delete cache, it's better use an option to don't use cache

Reply:
So, if we reach the 500 SurveyMonkey API calls, What you suggest to do?
The caching options are to minimize the amount of API calls, and the setting it's just a kill switch.

Reply:
tell me you, if surveymonkey_api_cache_duration = 0, the 500 api calls can be reached, and nobody knows that is a kill switch for you

Reply:
So, What you suggest to do? @andrey-canon

Reply:
I think you did this because you didn't want to wait for the cache when you were developing, for me, a normal user shouldn't have the ability to delete this, so  
1) Add a validation to doesn't allow values lower that 5 minutes for surveymonkey_api_cache_duration.
2) Add an option in studio to be disable the cache with a big message, something like "if you active this the cache wont be used and you could reach the max api calls ......"

Reply:
I did it by restarting the Memcached container...
And if I did some change in the SurveyMonkey, I need to get the fresh result and not the cache results.
All right I've added the message and about the first point, the implementation will remain ugly with that, so...